Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -87,7 +87,16 @@ class MessageQueueManagerImpl: MessageQueueManager {
     }
 
     func getInlineMessages(forElementId elementId: String) -> [Message] {
-        localMessageStore.filter { $0.value.elementId == elementId }.map(\.value).sortByMessagePriority()
+        let messages = localMessageStore.filter {
+            if $0.value.elementId == elementId {
+                if $0.value.doesHavePageRule() {
+                    return $0.value.doesPageRuleMatch(route: Gist.shared.getCurrentRoute())
+                }
+                return true
+            }
+            return false
+        }.map(\.value).sortByMessagePriority()
```

Comment:
```suggestion
        let messages = localMessageStore
            .map(\.value) // give us the Message from dictionary 
            .filter { $0.value.elementId == elementId } // Only get messages for a specific elementid. 
            .filter { $0.value.doesHavePageRule() && $0.value.doesPageRuleMatch(route: Gist.shared.getCurrentRoute()) } // if page rules enabled, filter what match             
            .sortByMessagePriority()
```

The logic looks good. I do find the code hard to read at the moment. I suggest using multiple filters, comments, or extensions to make this more usable.

Reply:
Thanks for the suggestion.
Small tweak was required in this suggestion. With the suggested logic, the message that matches the element id _and_ page rule were only filtered whereas the requirement is:

- Element id does not match then don't filter
- Element Id matches and page rule is inapplicable then filter
- Element Id matches and page rule is applicable and matches current route then filter
- Element Id matches and page rule is applicable but does not match current route then don't filter
 
 The suggestion was not satisfying the 2nd requirement and hence tweaks.